### DISCUSSION TITLE
Publish CycloneDX SBOM

### DISCUSSION THREAD

**Discussion by VinodAnandan** [fluxcd/flux2#2430] [Proposals]
CycloneDX is an OWASP flagship project ( https://owasp.org/www-project-cyclonedx/ ). The Open Web Application Security Project (https://owasp.org/) is a non-profit foundation that works to improve the security of software. CycloneDX is already supported by many security vendors and projects ( https://cyclonedx.org/about/supporters/ ). There are more than 100+ tools in the CycloneDX tool center - https://cyclonedx.org/tool-center/ . CycloneDX also supports Software-as-a-Service BOM (SaaSBOM), Operations Bill of Materials (OBOM), Vulnerability Exploitability Exchange (VEX), Bill of Vulnerabilities (BOV), Common Release Notes Format. etc. It is designed for use in application security contexts and supply chain component analysis.

cyclonedx-gomod ( https://github.com/CycloneDX/cyclonedx-gomod ) makes it easy to generate CycloneDX SBOM for Go applications. Example CycloneDX SBOMs ( https://github.com/CycloneDX/cyclonedx-gomod/tree/main/examples )

**Comment by kingdonb** [comment]
I would like to understand the reasons why Flux should support multiple SBOM formats, if there is a good answer to that question, as I don't currently have a solid understanding of the ways that SBOMs are used, or the difference between the formats.

I know that Flux has [recently added support for SPDX format SBOMs](https://github.com/fluxcd/flux2/issues/2302) and is publishing them with every release now.

I think this is something good to discuss at the Flux Community Meeting; we've put it on the agenda for tomorrow. See the calendar for information about how to join, (the times and meeting info are posted here, in UTC: https://fluxcd.io/#calendar)

Welcome to post back here if there is more information to add context, in case you won't be able to make it to the meeting!

**Reply by stefanprodan** [reply]
We've chosen SPDX because it's a Linux Foundation project same as Flux and Kubernetes. If Kubernetes switches to CycloneDX then we may consider it.

**Reply by stevespringett** [reply]
There are many LF projects that support CycloneDX in addition to SPDX including Buildpacks and Kyverno. Outside of LF projects, the majority of the security community uses CycloneDX. So if you want the SBOM to be useful outside of LF projects, it needs to support CycloneDX. But I think there's other opportunities here as well, like the ability to include services in the inventory of components, which is something SPDX is not capable of.

**Reply by VinodAnandan** [reply]
I would like to add that the Syft tool used by Flux for the SBOM generation is one of the early adopters of the CycloneDX; they have been supporting CycloneDX since 2020. Also, the author of the Kubernetes bom tool (@puerco) is planning to add CycloneDX support."

**Reply by VinodAnandan** [reply]
https://twitter.com/stevespringett/status/1508860883593838599
